tests/integration_tests/utils.py

```diff
@@ -683,3 +686,106 @@ def train_with_backend(
     finally:
         # Remove results/intermediate data saved to disk
         shutil.rmtree(output_dir, ignore_errors=True)
+
+
+# creates synthetic data and update feature definitions
+# This covers pre-processing functionality and
+# converting raw data to input specific tensors.
+def setup_input_feature_test(
+        batch_size: int = None,
+        feature_definition: Dict = None,
+        feature_class: Type[InputFeature] = None
+) -> Tuple:
+    # generate synthetic raw data for testing
+    np.random.seed(1919)
+    dataset = build_synthetic_dataset(
+        batch_size,
+        [feature_definition]
+    )
+    raw_data = '\n'.join([str(r[0]) for r in dataset])
+    df = pd.read_csv(io.StringIO(raw_data))
+
+    # preprocess raw data to create feature specific tensors and generate metadata
+    metadata = {}
+    input_data, _, _, metadata = DataFramePreprocessor.preprocess_for_training(
+        [feature_definition],
+        training_set=df,
+        training_set_metadata=metadata
+    )
+
+    # update feature definitions with defaults where required
```

Comment:
This is really useful, are the defaults specified in a config, if not does it make sense to move the defaults to config

Reply:
IIRC, the defaults are set as class attributes in each feature.